### PR TITLE
Dan bug smashing

### PR DESCRIPTION
**Pull request by DylanHsu**
Please hold off on merging

### CODE REVIEW

**Comment by dabercro** [comment]
There are also a bunch of selection cuts I want to talk about too. I'll keep a diff of what I've been fixing here: https://github.com/dabercro/hbb/pull/1/files#diff-81a29d3ae055ecc031f72cf8b4f16139

**Comment by dabercro** [comment]
Actually, I guess the conversion is a good place to go over it: https://github.com/dabercro/hbb/pull/1

**Comment by DylanHsu** [comment]
Loose jet ID added
Electron MVA preselection is added

The user can cut on electronSelBit & kEleMvaWP80 offline to count the leptons.

Still save all the electrons that pass the electron preselection (even if they are in the ecal crack)

The branch "nLooseLep" still counts the preselected electrons, it's the user's responsibility to use the right thing in their cut string.

**Comment by dabercro** [comment]
So this tight lepton cut came up already. (Leading electron is not tight, but sub-leading is) You should probably implement something for @maierbenedikt to get that right, besides throwing out the event.

**Comment by dabercro** [comment]
I'm actually having trouble throwing out the event properly either. If I try to check Muons and Electrons, it crashes on any empty vectors: 
```
root [18] events->Scan("nLooseMuon && muonSelBit[0]:electronSelBit[0]", "runNumber==273158 && lumiNumber==1050 && eventNumber==1488282179 && ((nLooseMuon && muonSelBit[0]&8) || (nLooseElectron && electronSelBit[0]&8))", "", 100000)
************************************
*    Row   * nLooseMuo * electronS *
************************************
************************************
==> 0 selected entries
(Long64_t) 0
root [19] events->Scan("nLooseMuon && muonSelBit[0]:electronSelBit[0]", "runNumber==273158 && lumiNumber==1050 && eventNumber==1488282179 && ((nLooseElectron && electronSelBit[0]&8))", "", 100000)
************************************
*    Row   * nLooseMuo * electronS *
************************************
*    90899 *         0 *        63 *
************************************
==> 1 selected entry
(Long64_t) 1
root [20] events->Scan("nLooseMuon && muonSelBit[0]:electronSelBit[0]", "runNumber==273158 && lumiNumber==1050 && eventNumber==1488282179 && ((nLooseElectron && electronSelBit[0]&8) || (nLooseMuon && muonSelBit[0]&8))", "", 100000)
************************************
*    Row   * nLooseMuo * electronS *
************************************
************************************
==> 0 selected entries
```

Note, the first and third attempt have electrons and muons switched in the cutstring.

File: Flat/src/ModulesLepPho.cc [hunk 1]
```diff
@@ -142,14 +142,20 @@ void PandaAnalyzer::ComplicatedLeptons() {
     bool isMedium = ele.medium;
     bool isTight  = ele.tight;
     bool isDxyz   = ElectronIP(ele.eta(),ele.dxy,ele.dz);
+    bool eleMVAPresel = 
+      pt>15 && ele.hOverE<0.09 && ele.combIso()/pt < 0.18 &&
+      (
+        (aeta<1.4442 && ele.sieie<0.012 && ele.ecalIso/pt < 0.25 && fabs(ele.dEtaInSeed)<0.0095 && fabs(ele.dPhiIn)<0.065) ||
+        (aeta>1.5660 && ele.sieie<0.033 && ele.ecalIso/pt < 0.45 && ele.hcalIso/pt<0.28)
+      );
```
Comment (dabercro) [inline Flat/src/ModulesLepPho.cc]: This doesn't exactly match the AN, but I wouldn't be surprised if you know more than them. Where is this from?

**Comment by DylanHsu** [comment]
Dan can you check now
https://github.com/sidnarayanan/PandaAnalysis/pull/74/commits/c42ad8c61ec1f6f5f4c834f1bc4e7733f54d8319

**Comment by dabercro** [comment]
I just realized I had swapped out trackiso for reliso, which is wrong. Thanks.

**Comment by DylanHsu** [comment]
plz merge

File: Flat/src/ModulesLepPho.cc [hunk 2]
```diff
@@ -183,15 +203,16 @@ void PandaAnalyzer::ComplicatedLeptons() {
     looseLeps.push_back(&ele);
     matchLeps.push_back(&ele);
     matchEles.push_back(&ele);
-    gt->nLooseElectron++;
+    // WARNING: The definition of "loose" here may not match your analysis definition of a loose electron for lepton multiplicity or jet cleaning considerations.
+    // It is the user's responsibility to make sure he is cutting on the correct multiplicity. Enough information is provided to do this downstream.
```
Comment (sidnarayanan) [inline Flat/src/ModulesLepPho.cc:207]: To be clear, this only applies to ComplicatedLeptons right?

Comment (DylanHsu) [reply Flat/src/ModulesLepPho.cc:207]: yes

**Comment by DylanHsu** [comment]
please merge

**Pull request merged**